pygaggle/data/segmentation.py

```diff
@@ -0,0 +1,47 @@
+import spacy
+import numpy as np
+from pygaggle.rerank.base import Text
+from typing import List
+
+
+class SegmentProcessor:
+
+    def __init__(self, documents: List[Text]):
+        self.doc_end_indexes = [0]
+        self.documents = documents
+        self.aggregate_methods = {
+            "max": self._max_aggregate,
+            "mean": self._mean_aggregate
+        }
+
+    def segment(self, seg_size: int, stride: int) -> List[Text]:
+        segmented_doc, end_idx = [], 0
+        nlp = spacy.blank("en")
+        nlp.add_pipe(nlp.create_pipe("sentencizer"))
+        for document in self.documents:
+            doc = nlp(document.text[:1000000])
+            sentences = [sent.string.strip() for sent in doc.sents]
+            for i in range(0, len(sentences), stride):
+                segment_text = ' '.join(sentences[i:i + seg_size])
+                segmented_doc.append(Text(segment_text))
+                if i + seg_size >= len(sentences):
+                    end_idx += i/stride + 1
+                    self.doc_end_indexes.append(int(end_idx))
+                    break
+        return segmented_doc
+
+    def aggregate(self, scores: List[int], method: str = "max"):
+        for i in range(len(self.documents)):
+            scores_start_idx = self.doc_end_indexes[i]
```

Comment:
Could you please add a comment explaining why we need `self.doc_end_indexes`?

Reply:
```'self.doc_end_indexes' will store the index of the last segment of each document when 'segment()' converting a 'List[Text]' of documents into 'List[Text]' of segments. It will be used to split and group segments' scores and feed the aggregated scores back to documents in 'aggregate()'```

Will add the comment into the file.

Reply:
Great, thanks!